zerver/lib/streams.py

```diff
@@ -411,10 +410,13 @@ def check_stream_access_for_delete_or_update(
     if user_profile.is_realm_admin:
         return
 
+    if can_administer_channel(stream, user_profile, sub):
+        return
```

Comment:
Issue mentions that only subscribers can change permission and add users to the channel. I guess controlling adding users is not part of the PR currently, but we may need subscriber check for changing permissions.

Issue does not mention anything about changing stream privacy (public, web-public or private), so not sure about that. We already have restriction of allowing only subscribed admins to make a private stream public.

Reply:
Yeah I think the correct strategy is to have this mirror the realm admin check in terms of where it's placed. 

I guess an abstractions question is whether it'd be better to make `can_administer_channel` be the place that `user_profile.is_realm_admin` gets checked as the refactoring strategy here. I think that'd be preferable for this code path? Would reduce the number of duplicate copies of authorization logic that we have.

Reply:
There might be still a follow-up around the abstractions to track here?

Reply:
Addressed in #32558 (6th commit)